### ISSUE TITLE
[Feature] Add accuracy metric for QNN

### ISSUE THREAD

**Issue by DanieleCucurachi** [pasqal-io/qadence#521]
At the moment, the only built-in metric for evaluating the quality of the learned solution in the context of QNNs is the training loss.

While a learned solution with zero loss will match the exact solution, the quality of solutions with small but non-zero losses (a more realistic scenario) remains uncertain as common choices like $L_2$ and $L_1$ losses do not provide any accuracy guarantees.

**Proposed Feature:** Introduce the possibility to calculate and log the model's error against a known benchmark (ground truth). This will provide a more comprehensive evaluation of the learned solution’s quality.

**Comment by chMoussa** [comment]
@DanieleCucurachi To try with the new callback features in TrainConfig #533
